collector_test.go

```diff
@@ -2,26 +2,119 @@ package main
 
 import (
 	"reflect"
+	"regexp"
 	"testing"
 
+	"github.com/prometheus/client_model/go"
 	"github.com/soniah/gosnmp"
 
-	"github.com/prometheus/client_model/go"
 	"github.com/prometheus/snmp_exporter/config"
 )
 
 func TestPduToSample(t *testing.T) {
+	var regexpMatch, regexpNoMatch config.Regexp
+	regexpMatch.Regexp, _ = regexp.Compile(".*")
```

Comment:
Put these inline in the struct, it's easier to understand the tests that way. You can use MustCompile